oak_attestation_verification/src/verifier.rs

```diff
@@ -394,6 +394,14 @@ fn verify_amd_sev_attestation_report(
         anyhow::bail!("debug mode not allowed");
     }
 
+    if reference_values
+        .firmware_version
+        .as_ref()
+        .is_some_and(|a| a.values.len() > 0)
+    {
+        anyhow::bail!("firmware version check needs implementation");
```

Comment:
Is this the only thing needing implementation? How about the signature of the attestation itself?

Reply:
verify_attestation_report_signature verifies the signature, this is just about the contents of the attestation report.

Reply:
What about the stage0 field?

Reply:
the initial measurement (that implicity includes Stage0) is in the attestation report as well, so probably requires similar handling here as well.

Reply:
We don't have the ingredients ready to do this. I suggest we first work on derived endorsements. For stage0 all we could do here is to verify endorsements against reference values (and ignoring the evidence). WDYT?

Reply:
Added TODO and resolving. The stage0 derived measurement (initial_measurement) is a SHA2-384 hash which we can store in a RawDigest to compare against the endorsement. Then it will be a straightforward call to verify_measurement_digest().

Reply:
Contrary to what I wrote before, the stage0 case is now addressed: we compare it as if the derived endorsement were already available. Since it is not available, the stage0 verification fails but can (and needs to be) skipped. This is the only way to make the verification pass.

Added unit tests (a) for fail with set firmware version ref value, (b) for fail with set stage0 ref value. These unit tests are short lived.

See discussion in b/327069120.